lib/sedna/algorithms/multi_task_learning/multi_task_learning.py

```diff
@@ -36,130 +36,126 @@ class MulTaskLearning:
 
     def __init__(self,
                  estimator=None,
-                 task_definition="TaskDefinitionByDataAttr",
+                 task_definition=None,
                  task_relationship_discovery=None,
                  task_mining=None,
                  task_remodeling=None,
-                 inference_integrate=None,
-                 task_definition_param=None,
-                 relationship_discovery_param=None,
-                 task_mining_param=None,
-                 task_remodeling_param=None,
-                 inference_integrate_param=None
+                 inference_integrate=None
                  ):
-        if not task_relationship_discovery:
-            task_relationship_discovery = "DefaultTaskRelationDiscover"
-        if not task_remodeling:
-            task_remodeling = "DefaultTaskRemodeling"
-        if not inference_integrate:
-            inference_integrate = "DefaultInferenceIntegrate"
-        self.method_selection = dict(
-            task_definition=task_definition,
-            task_relationship_discovery=task_relationship_discovery,
-            task_mining=task_mining,
-            task_remodeling=task_remodeling,
-            inference_integrate=inference_integrate,
-            task_definition_param=task_definition_param,
-            task_relationship_discovery_param=relationship_discovery_param,
-            task_mining_param=task_mining_param,
-            task_remodeling_param=task_remodeling_param,
-            inference_integrate_param=inference_integrate_param)
+
+        self.task_definition = task_definition or {
+            "method": "TaskDefinitionByDataAttr"
+        }
+        self.task_relationship_discovery = task_relationship_discovery or {
+            "method": "DefaultTaskRelationDiscover"
+        }
+        self.task_mining = task_mining or {}
+        self.task_remodeling = task_remodeling or {
+            "method": "DefaultTaskRemodeling"
+        }
+        self.inference_integrate = inference_integrate or {
+            "method": "DefaultInferenceIntegrate"
+        }
         self.models = None
         self.extractor = None
         self.base_model = estimator
         self.task_groups = None
-        self.task_index_url = Context.get_parameters(
-            "MODEL_URLS", '/tmp/index.pkl'
-        )
+        self.task_index_url = KBResourceConstant.KB_INDEX_NAME.value
         self.min_train_sample = int(Context.get_parameters(
```

Comment:
Same, comments are needed

Reply:
Currently we do not regard the knowledge base as a public interface. Of course, it should also be added in the PR of document

Reply:
Well, this comment is talking about the "Context.get_parameters", not the knowledge base. 

Good to learn that it will be followed up. Also, it would be great if we have related issues in the community.